thrift/thrift-gen/template.go

```diff
@@ -29,14 +29,42 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os/exec"
+	"strings"
 	"text/template"
 )
 
-func parseTemplate() *template.Template {
+// Template represents a single thrift-gen template that will be used to generate code.
+type Template struct {
+	name     string
+	template *template.Template
+}
+
+func parseTemplate(contents string) (*template.Template, error) {
 	funcs := map[string]interface{}{
-		"contextType": contextType,
+		"contextType":  contextType,
+		"lowercase":    strings.ToLower,
```

Comment:
Is that the intent for this function? In that case, we definitely shouldn't use `lowerCase`, but I can put in `goPrivateName` which will actually point to the function `goName` in `names.go`. That way, you can pass in names that are reserved keywords and it'll automatically add a suffix.